core/Cargo.toml

```diff
@@ -38,6 +38,8 @@ rlp_derive = { git = "https://github.com/CodeChain-io/rlp.git", version = "0.2"
 snap = "0.2"
 table = { path = "../util/table" }
 util-error = { path = "../util/error" }
+num-traits = "0.2"
```

Comment:
I couldn't agree more changing constants to enum is good, but I have a question whether using num-traits and num-derive or not.
I think implementing `RlpEncodable` and `RlpDecodable` instead of using them seems good either.
How about you @majecty?

Reply:
It seems good to use `RlpEncodable` and `RlpDecodable`. @somniumism Could you implement Rlp encoding/decoding functions for the enum type?

Reply:
What the `num-traits` is doing is that creating a conversion function from a number type to an enum type. The conversion function is required in the serialize/deserialize process.

If we implement `Encodable` trait and `Decodable` trait to the enum types, we don't need to cast. The Rlp code will receive and return enum type values directly.